### PR TITLE
Adding eliminating-null-parameters migration guide

### PR DESCRIPTION
**Pull request by gspencergoog**
Adding a migration guide for eliminating null parameters.

### CODE REVIEW

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+## Summary
+
+This migration guide describes conversion of code that uses the `nullOk`
+parameter on multiple `of` static accessors and related accessors to use
+alternate APIs with nullable return values.
+
+## Context
+
+Flutter has a common pattern of allowing lookup of some types of widgets
+(`InheritedWidget`s) using static member functions that are typically called
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:15]: nit: maybe make InheritedWidget a link to API docs?

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+`of`, and take a `BuildContext`.
+
+Before non-nullability was the default, it was useful to have a toggle on these
+APIs that swapped between throwing an exception if the widget was not present in
+the widget tree and returning null if it was not found. It was useful, and
+wasn't confusing, since every variable was nullable.
+
+When non-nullability was made the default, it was then desirable to have the
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:23]: nit: maybe break this paragraph up into multiple sentences to improve readability?

**Review by goderbauer — APPROVED** [review]
LGTM

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 1 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md]: Done.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 2 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:23]: Run-on sentence much? Yeah, I split it into multiple sentences and multiple paragraphs.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+1. [`ScaffoldMessenger.of`]
+1. [`Scaffold.of`]
+1. [`Router.of`]
+1. [`Localizations.localeOf`]
+1. [`FocusTraversalOrder.of`]
+1. [`FocusTraversalGroup.of`]
+1. [`Focus.of`]
+1. [`Shortcuts.of`]
+1. [`Actions.handler`]
+1. [`Actions.find`]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:58]: Actions.invoke is missing here

**Review by Piinks — CHANGES_REQUESTED** [review]

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+1. [`Actions.maybeHandler`]
+1. [`Actions.maybeFind`]
+1. [`Actions.maybeInvoke`]
+1. [`AnimatedList.maybeOf`]
+1. [`SliverAnimatedList.maybeOf`]
+1. [`CupertinoDynamicColor.maybeResolve`]
+1. [`CupertinoDynamicColor.maybeResolveFrom`]
+1. [`CupertinoUserInterfaceLevel.maybeOf`]
+1. [`CupertinoTheme.maybeBrightnessOf`]
+1. [`CupertinoThemeData.maybeResolveFrom`]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:92]: I have not been able to find this method on master.

Comment (Piinks) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:92]: This one may be missing.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+1. [`NoDefaultCupertinoThemeData.maybeResolveFrom`]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:93]: Likewise here.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+1. [`CupertinoTextThemeData.maybeResolveFrom`]
+1. [`MaterialBasedCupertinoThemeData.maybeResolveFrom`]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:95]: These two as well.

**Review by Piinks — COMMENTED** [review]
To sum up, these methods appear to be missing:
- `CupertinoThemeData.maybeResolveFrom`
- `NoDefaultCupertinoThemeData.maybeResolveFrom`
- `CupertinoTextThemeData.maybeResolveFrom`
- `MaterialBasedCupertinoThemeData.maybeResolveFrom`

Updated after audit:
- `Actions.maybeHandler`
- `CupertinoDynamicColor.maybeResolveFrom`

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+1. [`Navigator.maybeOf`]
+1. [`ScaffoldMessenger.maybeOf`]
+1. [`Scaffold.maybeOf`]
+1. [`Router.maybeOf`]
+1. [`Localizations.maybeLocaleOf`]
+1. [`FocusTraversalOrder.maybeOf`]
+1. [`FocusTraversalGroup.maybeOf`]
+1. [`Focus.maybeOf`]
+1. [`Shortcuts.maybeOf`]
+1. [`Actions.maybeHandler`]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:83]: Running an audit to double check from last night, I can't find Actions.maybeHandler either

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+1. [`Actions.maybeFind`]
+1. [`Actions.maybeInvoke`]
+1. [`AnimatedList.maybeOf`]
+1. [`SliverAnimatedList.maybeOf`]
+1. [`CupertinoDynamicColor.maybeResolve`]
+1. [`CupertinoDynamicColor.maybeResolveFrom`]
```
Comment (Piinks) [inline src/docs/release/breaking-changes/eliminating-nullok-parameters.md:89]: This one too.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 7 above)
Comment (Piinks) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:83]: Looks like this is intentional per the change description: https://github.com/flutter/flutter/pull/68921
```suggestion
```

**Comment by Piinks** [comment]
Sorry for flooding your inbox, I think I understand that the 'missing' methods may not be missing, but instead are listed here by mistake. The PR description in https://github.com/flutter/flutter/pull/68905 mentions not needing the maybeOf for all of them

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 3 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:58]: Added.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 7 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:83]: Yes, removed.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 8 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:89]: Oh, yeah, we did remove that one!

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 4 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:92]: Yep.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 5 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:93]: Removed.

File: src/docs/release/breaking-changes/eliminating-nullok-parameters.md (hunk 6 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/eliminating-nullok-parameters.md:95]: Yep

**Review by gspencergoog — COMMENTED** [review]
OMG. Thanks: I wrote this when we started, and a bunch of things changed along the way, I guess. I think we decided that all the `resolveFrom` routines should never return null anyhow, and just eliminated the maybeResolveFrom versions. Anyhow, thanks for going through those.

**Review by Piinks — APPROVED** [review]

**Pull request merged**